utils/gitutils/gitmanager_test.go

```diff
@@ -0,0 +1,138 @@
+package gitutils
+
+import (
+	"path/filepath"
+	"testing"
+
+	goGit "github.com/go-git/go-git/v5"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	securityTestUtils "github.com/jfrog/jfrog-cli-security/tests/utils"
+
+	"github.com/jfrog/jfrog-client-go/xsc/services"
+)
+
+func TestGetGitContext(t *testing.T) {
+	basePath := filepath.Join("..", "..", "tests", "testdata", "git", "projects")
+
+	testCases := []struct {
+		name                  string
+		testProjectZipDirPath string
+		NoDotGitFolder        bool
+		gitInfo               *services.XscGitInfoContext
+	}{
+		{
+			name:                  "No Git Info",
+			NoDotGitFolder:        true,
+			testProjectZipDirPath: filepath.Join(basePath, "nogit"),
+		},
+		{
+			name:                  "Clean Project (after clone)",
+			testProjectZipDirPath: filepath.Join(basePath, "clean"),
+			gitInfo: &services.XscGitInfoContext{
+				GitRepoHttpsCloneUrl: "https://github.com/attiasas/test-security-git.git",
+				GitRepoName:          "test-security-git",
+				GitProject:           "attiasas",
+				GitProvider:          "github",
+				LastCommitHash:       "5fc36ff0666e5ce9dba6c0a1c539ee640cabe0b0",
+				LastCommitMessage:    "remove json",
+				LastCommitAuthor:     "attiasas",
+
+				BranchName: "main",
+			},
+		},
+		{
+			name:                  "Self-Hosted Git Project (and SSO credentials)",
+			testProjectZipDirPath: filepath.Join(basePath, "selfhosted"),
+			gitInfo: &services.XscGitInfoContext{
+				GitRepoHttpsCloneUrl: "ssh://git@git.jfrog.info/~assafa/test-security-git.git",
```

Comment:
In Xray, the ssh clone url doesn't have ~ in the URL:
`ssh://git@git.jfrog.info/xray/xray.git`

Reply:
it may have. this test is testing for that.